auto-dark.el

```diff
@@ -53,11 +57,21 @@ to check dark-mode state, if `ns-do-applescript' is not available."
   :group 'auto-dark
   :type 'boolean)
 
+(defcustom auto-dark-detection-method nil
+  "The method auto-dark should use to detect the system theme.
+
+Defaults to nil and will be populated through feature detection
+if left as such.  Only change this value if you know what you're
+doing!"
+  :group 'auto-dark
+  :type 'symbol
+  :options '(applescript osascript dbus powershell))
+
 (defvar auto-dark--last-dark-mode-state 'unknown)
 
 (defvar auto-dark--dbus-listener-object nil)
 
-(defun auto-dark--is-dark-mode-builtin ()
+(defun auto-dark--is-dark-mode-applescript ()
```

Comment:
Given that this library now supports other OSs, and Emacs' `dbus` library is built-in, "builtin" was a bit ambiguous.